src/git/git.go

```diff
@@ -53,6 +72,345 @@ func (g *GitVersioning) validate() error {
+// Returns:
+// 		error: Returns an error whenever unexpected issues happen.
+func (g *GitVersioning) addToStage() error {
+	worktree, err := g.repo.Worktree()
+	if err != nil {
+		return err
+	}
+
+	err = worktree.AddWithOptions(&git.AddOptions{All: true})
+	if err != nil {
+		return err
+	}
+	g.log.Info(colorGreen + "Changes added to stage area..." + colorReset)
+	return nil
+}
+
+// commitChanges aims to commit all the staging changes to the transfer area.
+// Args:
+// 		newReleaseVersion (string): The new release version. I.e.: "1.0.1".
+// Returns:
+// 		error: Returns an error whenever unexpected issues happen.
+func (g *GitVersioning) commitChanges(newReleaseVersion string) error {
+	if err := g.addToStage(); err != nil {
+		return err
+	}
+
+	worktree, err := g.repo.Worktree()
+	if err != nil {
+		return err
+	}
+
+	signature := &object.Signature{Name: g.mostRecentCommit.authorName, Email: g.mostRecentCommit.authorEmail, When: time.Now()}
+
+	message := fmt.Sprintf("type: [skip]: message: Commit automatically generated by Semantic Release. The new tag is %s", newReleaseVersion)
+	commit, err := worktree.Commit(message, &git.CommitOptions{Author: signature, Committer: signature})
+	if err != nil {
+		return err
+	}
+
+	g.log.Info(colorGreen+"New commit added: %s"+colorReset, commit.String())
+	return nil
+}
+
+// push aims to push the commits to the remote repository.
+// Returns:
+// 		error: Returns an error whenever unexpected issues happen.
+func (g *GitVersioning) push() error {
+	err := g.repo.Push(&git.PushOptions{
+		Auth: &http.BasicAuth{
+			Username: g.username,
+			Password: g.password,
+		},
+		InsecureSkipTLS: true})
+	if err != nil {
+		return fmt.Errorf("error while pushing commits to remote repository due to: %s", err.Error())
+	}
+
+	return nil
+}
+
+// tagExists aims to find a given tag in the repository.
+// 		tag (string): Tag name to search.
+// Returns:
+// 		bool: True when tag found, otherwise false.
+// 		error: Returns an error whenever unexpected issues happen.
+func (g *GitVersioning) tagExists(tag string) (bool, error) {
+	res := false
+	tags, err := g.repo.TagObjects()
+	if err != nil {
+		return res, fmt.Errorf("error while getting tags due to %s", err.Error())
+	}
+
+	err = tags.ForEach(func(t *object.Tag) error {
+		if t.Name == tag {
+			res = true
+			return fmt.Errorf("tag %s already exists", tag)
+		}
+		return nil
+	})
+	if err != nil && err.Error() != fmt.Sprintf("tag %s already exists", tag) {
+		return false, fmt.Errorf("error while iterating tags due to: %s", err)
+	}
+	return res, nil
+}
+
+// setTag aims to create a given tag in the local repository.
+// Args:
+// 		tag (string): Tag name to be created.
+// Returns:
+// 		err: Error whenever unexpected issues happen.
+func (g *GitVersioning) setTag(tag string) error {
+	g.log.Info("Set tag %s", tag)
+	tagExists, err := g.tagExists(tag)
+	if err != nil {
+		return err
+	}
+	if tagExists {
```

Comment:
if tag exist, shouldn't you return one error? if you don't stop release process.. we can replace some exist docker image for example... @esequielvirtuoso

Reply:
Yep, I'll change it.